hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/HazelcastResources.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.calcite.validate;
+
+import org.apache.calcite.runtime.CalciteException;
+import org.apache.calcite.runtime.Resources;
+import org.apache.calcite.sql.validate.SqlValidatorException;
+
+/**
+ * Error messages for parsing and validation stages.
+ */
+public interface HazelcastResources {
+
+    HazelcastResources RESOURCES = Resources.create(HazelcastResources.class);
+
+    @Resources.BaseMessage("Function ''{0}'' does not exist")
+    Resources.ExInst<CalciteException> functionDoesNotExist(String functionName);
+
+    @Resources.BaseMessage("Cannot apply {1} to the {0} operator (consider adding an explicit CAST)")
+    Resources.ExInst<SqlValidatorException> invalidOperatorOperands(String operatorName, String operandTypes);
+
+    @Resources.BaseMessage("Cannot apply {1} to the {0} function (consider adding an explicit CAST)")
+    Resources.ExInst<SqlValidatorException> invalidFunctionOperands(String functionName, String operandTypes);
```

Comment:
Shouldn't it be in reverse? Can't apply operator/function to operands, not the operands to operator/function.

Reply:
Fixed in https://github.com/hazelcast/hazelcast/pull/17947/commits/ba60023f747f99fd5df2e218577873f153c071d4